app/javascript/src/component_expander.js

```diff
@@ -1,111 +1,127 @@
 /**
- * Dialog Component
+ * Expander Component
  * ----------------------------------------------------
  * Description:
- * Creates an area that expands and contracts, activated by a single element.
- *
- * Documentation:
- *
- *     - jQueryUI
- *       https://api.jqueryui.com/dialog
- *
- *     - TODO:
- *       (steven.burnell@digital.justice.gov.uk to add).
+ * Creates an area that expands and contracts, activated by a single title element.
  *
+ * Implements the following behaviour:
+ *  - Takes a provided element, finds the configured title element and enhances
+*  it by wrapping the contents with a <button> activator.
+*   - If config.wrap_content is true, it will wrap all other child elements
+*   within the provided element in a <div>
+*   - By default, it will collapse and hide the content.
+*
+*   Example:
+*   Given the following:
+*   <dl>
+*     <dt>Title</dt>
+*     <dd>Loem ipsum dolor sit amet consecteteur adipiscing elit...</dd>
+*   </dl>
+*   <script>
+*     var $list = $(dl);
+*     new Expander($list, {
+*       title_tag: 'dt',
+*       wrap_content: false,
+*     })
+*   </script>
+*
+*   Would result the following markup:
+*   <dl class="Expander close">
+*     <dt class="Expander__title">
+*       <button aria-expanded="false" aria-controls="Expander_12345678790">Title</button>
+*     </dt>
+*     <dd id="Expander_1234567890" class="Expander__container" style="display: none;">
+*       Lorem ipsum.....
+*     </dd>
+*   </dt>
  **/
 
-const utilities = require('./utilities');
+const { 
+  mergeObjects,
+  uniqueString
+} = require('./utilities');
 
-
-/* Main class
- * Select an element that should show/hide its content based on activator toggle.
- * The component will inject a container that wraps the elements content.
- * An activator will be created, if not specified, and moved to be placed just
- * before the injected container element. Toggle of the activator will show/hide
- * the content, which is now wrapped by the injected container.
- *
- * @$node  (jQuery node) Element found in template that should be enhanced.
- * @config (Object) Configurable key/value pairs.
- **/
+/**
+ * @param $node (jQuery) Html Element to be enhanced.
+ * @param condif (Object) Key/value pairs for config (see above)
+**/
 class Expander {
+  #config;
+
   constructor($node, config) {
-    var conf = utilities.mergeObjects({
-      activator_text: "Toggle", // Text for any self-created activator button.
-      auto_open: false, // Set whether open on creation.
-      duration: 0, // Number determining how long the animation will run
-      $activator: null // Pass in the jQuery element you want to toggle open/close.
+    const conf = mergeObjects({
+      title_tag: 'h2',
+      wrap_content: false,
```

Comment:
I agree the default should probably be `false` - I had it setup like this so that the defaults were deliberately what I didn't want, in order to be sure that overriding them worked.